libdevcore/Algorithms.h

```diff
@@ -67,8 +67,8 @@ class CycleDetector
 
 private:
 	std::function<void(V const&, CycleDetector&)> m_visit;
-	std::set<V const*> m_processing;
-	std::set<V const*> m_processed;
+	std::set<V const*> m_processing{};
```

Comment:
Oh wow, this is really required? What is the idea behind?

Reply:
Another option would be to really initialize these members in the constructor initialization lists..

Reply:
@chfast Thx for your feedback! With this PR I just wanted to get rid of the `-Weffc++` warnings the easy way.. but it was basically not possible to do that in a clean way because it looks like that they didn't take into account any context... However, somehow the current PR groups all member-initialization related things.. I used it to review the current member-initialization code (so the old code - not the new code).. I was not able to find uninitialized members in the current code (the original code currently used, not the actually changes of this PR)..